### PR TITLE
fix: Updated integration to allow for paginated ListPolicies

### LINKED ISSUE #357: Facing a cloudformation error when running servlerless package
```
{"code":"AWS_CLOUD_FORMATION_CREATE_STACK_ALREADY_EXISTS_EXCEPTION","providerError":{"message":"Stack [NewRelicLicenseKeySecret] already exists","code":"AlreadyExistsException","time":"2023-06-09T02:12:50.121Z","requestId":"23fa696b-90ed-4f9c-bb17-93c9d9beebd3","statusCode":400,"retryable":false,"retryDelay":82.09528233197582},"providerErrorCodeExtension":"ALREADY_EXISTS_EXCEPTION"}
```

Not sure if this is expected behaviour when running `sls package`

### PR DESCRIPTION
**Pull request by mrickard**
Customers with more than 100 locally-scoped IAM policies could encounter an error in which the plugin tries to recreate the secret access policy. This change follows the ListPolicies pagination standard. 

Closes #357 
Closes NR-154677

### CODE REVIEW

File: src/integration.ts [hunk 1]
```diff
@@ -86,24 +86,39 @@ export default class Integration {
     );
   }
 
+  public async makePaginatedRequest(params, regionFilter) {
+    const results = await this.awsProvider.request(
```
Comment (bizob2828) [inline src/integration.ts:90]: do we want a test that verifies it can paginate accordingly?

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: src/integration.ts (hunk 1 above)
Comment (mrickard) [reply src/integration.ts:90]: Let me see if I can mock that out.

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: tests/integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    awsProvider,
+    serverless,
+    region: "us-east-1",
+    licenseKey: "nr-license-key",
+    log: logShim,
+  };
+
+  const slsIntegration = new Integration(pluginMock);
+
+  describe("checkForManagedSecretPolicy makes a ListPolicies request", () => {
```
[the remark this answers is not in the document]
Comment (bizob2828) [reply tests/integration.test.ts:47]: also where makePaginatedRequest fails and logs error and returns empty object

[the remark this answers is not in the document]
Comment (mrickard) [reply tests/integration.test.ts:56]: @bizob2828 Those tests are now added.

**Review by bizob2828 — DISMISSED** [review]

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**